File: frontend/src/app/models/new-trip-info.ts
import { TimeInterval } from "./time-interval";

export class NewTripInfo {
    planID?: number;
    tripName!: string;
    startDate!: Date;
    endDate!: Date;
    range!: number;
    startLocation!: google.maps.LatLngLiteral;;
    tripTimeSlots!: TimeInterval[][];
  }

File: backend/src/main/java/com/demo/backend/models/DTO/NewTripInfo.java
package com.demo.backend.models.DTO;

import com.google.maps.model.LatLng;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
public class NewTripInfo {
    private String tripName;
    private Date startDate;
    private Date endDate;
    private int range;
    private LatLng startLocation;
    private List<List<TimeInterval>> tripTimeSlots;
}
